site/src/lib/api/blocks/npm.ts

```diff
@@ -30,7 +29,7 @@ const mirrorNpmPackageToR2 = async ({
   expandedMetadata: ExpandedBlockMetadata;
 }> => {
   const { path: npmTarballFolder, cleanup: cleanupDistFolder } = await tmp.dir({
-    tmpdir: isRunningOnVercel ? "/tmp" : undefined, // Vercel allows limited file system access
+    tmpdir: os.tmpdir(),
```

Comment:
I created a new `/api/tmpdir` endpoint in https://github.com/blockprotocol/blockprotocol/pull/565/commits/cafa76b060637733e330a2c2f067ea20e328f716. Result: https://blockprotocol-a9i8sizgj.stage.hash.ai/api/tmpdir → `/tmp`. This matches what a member of Vercel has publicly stated in https://github.com/vercel/vercel/discussions/4606#discussioncomment-24540.

Which flow was failing for you? Was it in Lambdas or could it be during Vercel build somehow?

I've removed `tmpdir: os.tmpdir()` in https://github.com/blockprotocol/blockprotocol/pull/565/commits/c93d0f5c0798103687ca745c93c635c746eeb297 – happy to test everything once we've updated env vars on Vercel.

Reply:
It was in the lambdas – if Vercel is returning `/tmp` for `os.tmpdir()` then it should work omitting it, since the library uses that by default. I'm not sure why I thought specifying `/tmp` rather than `os.tmpdir()` was necessary, may have been an unrelated but nearby error.